template/rstudio.script.sh

```diff
@@ -7,7 +7,7 @@ export PATH=/usr/lib/rstudio-server/bin:$PATH
 RSERVER_ARGLIST=(
   "--server-working-dir" "${WORKING_DIR}"
   "--server-user" "$(whoami)"
-  "--server-data-dir" "/tmp/run"
+  "--server-data-dir" "/tmp/$(whoami)/run"
```

Comment:
Based on testing, it looks like these changes work well (not getting any errors). I had to read [Apptainer Bind Paths and Mounts](https://apptainer.org/docs/user/main/bind_paths_and_mounts.html) to fully grok that `/tmp` is one of the default bind mounts (for some reason that didn't click).

Two outstanding questions I would have if this approach doesn't work as expected:
1. Is it desirable to include the `session.id` so that if a user runs multiple instances on the same compute node, there's no conflict? This may be a non-issue in practice.
2. Is there any benefit to re-binding `/tmp` in the container so that *any* use of that folder will automatically be separated per user? 

For example, we could do something along these lines:

```
TMP_DIR=/tmp/$USER/<%= session.id %>
mkdir -m 700 -p $TMP_DIR

SING_BIND_LIST=(
  "-B" "/shared/courseSharedFolders"
  "-B" "/shared/spack"
  "-B" "$WORKING_DIR/etc/rstudio:/etc/rstudio"
  "-B" "$TMP_DIR:/tmp"
)
```

One advantage to that is that it would mean the `database.conf.erb` could just set the directory to `/tmp/lib/rstudio-server` and it would end up writing to `/tmp/{user}/{session.id}/lib/rstudio-server` as a result of the bind mount. And any other temp folder usage would similarly be isolated.

Reply:
Hm, interesting. Changing that bind mount might be a way to reduce the number of places that we need to change or update something if there's any issue with the /tmp directory. That's a change that would be good to work into a future update. Thanks!